BLSQ/iaso
# IA-3637_IA-3638: User and User role edition: Implement dependency between Org unit management permission and Org unit write type

What problem is this PR solving? Explain here in one sentence.
- User edition: Implement dependency between Org unit management permission and Org unit write type
- User role edition: Implement dependency between Org unit management permission and Org unit write type
Related JIRA tickets : [IA-3637](https://bluesquare.atlassian.net/browse/IA-3637) [IA-3638](https://bluesquare.atlassian.net/browse/IA-3638)

## Self proofreading checklist


## Doc

Tell us where the doc can be found (docs folder, wiki, in the code...).

## Changes
- Added a tooltip and disable on the tab corresponding to the "Org Unit Write Types" both on user and user role createEdit form

## How to test
- Create or update a user or a user role and remove from him **Organisation units management - write** permission
- Check if the **Org Unit Write Types** tab is disables

## Print screen / video
[Screencast from 2024-11-20 08-29-39.webm](https://github.com/user-attachments/assets/39ab4522-1d38-4494-a5c6-7efab3b025cb)
[Screencast from 2024-11-20 08-30-13.webm](https://github.com/user-attachments/assets/f6f2a8d8-bd9f-4255-8e85-8962f38e4ca6)

## Notes

Things that the reviewers should know:

- known bugs that are out of the scope of the PR
- other trade-offs that were made
- does the PR depends on a PR in [bluesquare-components](https://github.com/BLSQ/bluesquare-components)?
- should the PR be merged into another PR?

## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: empty instance pop up

Refs: IA-3665
```

Note that the Jira reference is preceded by a _line break_.

Both the line break and the Jira reference are entered in the _Add an optional extended description…_ field.


[IA-3637]: https://bluesquare.atlassian.net/browse/IA-3637?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[IA-3638]: https://bluesquare.atlassian.net/browse/IA-3638?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
